dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/persistence/file/batch/BatchFileDataWriter.kt

```diff
@@ -68,39 +83,14 @@ internal open class BatchFileDataWriter<T : Any>(
         val byteArray = serializer.serializeToByteArray(data, internalLogger) ?: return
 
         synchronized(this) {
-            val success = writeData(byteArray)
-            if (success) {
-                onDataWritten(data, byteArray)
-            } else {
-                onDataWriteFailed(data)
+            val context = contextProvider.context
+            val eventId = System.identityHashCode(data).toString()
+            storage.writeCurrentBatch(context) {
+                pendingWriteEvents[eventId] = data to byteArray
```

Comment:
aren't we concerned that we will keep too much data in HEAP ?

Reply:
Good question. For now we need to store both event and serialized event mostly because of the RUM, but maybe later on with once RUM will start using write scope to perform writes we can change it.

What is your expectation for the memory pressure during SR? How often and how big are going to be events received? I guess concern is mostly because of the bitmaps?

I would propose to add TODO for now. WDYT?

Reply:
Yes I am mostly thinking for Bitmaps here, let's benchmark this when we get there and see how it goes.